sonosco/datasets/download_datasets/an4.py

```diff
@@ -0,0 +1,108 @@
+import os
+import click
+import io
+import shutil
+import tarfile
+import logging
+import sonosco.common.audio_tools as audio_tools
+import sonosco.common.path_utils as path_utils
+
+from sonosco.datasets.download_datasets.data_utils import create_manifest
+from sonosco.common.utils import setup_logging
+from sonosco.common.constants import *
+
+LOGGER = logging.getLogger(__name__)
+
+AN4_URL = 'http://www.speech.cs.cmu.edu/databases/an4/an4_raw.bigendian.tar.gz'
+
+def try_download_an4(target_dir, sample_rate, min_duration, max_duration):
```

Comment:
what do you mean intelligently? What I would like to have is a global function which is currently just a stub in the download_datasets/__init__.py download_all_datasets which would check which datasets are currently present and then download the missing ones and create a unified manifest. But I don't really see a problem with having a try_download_* as it currently is because after checking if the specified dataset is not present we can just call this function to download it.